core/common/src/MapLayerSettings.ts

```diff
@@ -3,21 +3,39 @@
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
 /** @packageDocumentation
- * @module DisplayStyles
+ * @module MapLayers
  */
 
 import { assert, Id64String } from "@itwin/core-bentley";
 import { BackgroundMapProvider, BackgroundMapProviderProps, BackgroundMapType } from "./BackgroundMapProvider";
 import { DeprecatedBackgroundMapProps } from "./BackgroundMapSettings";
 
-/** @beta */
+/** Enum of imagery map layer formats Ids supported.
+ * @public
+ */
+export enum ImageryMapLayerFormatId {
+  ArcGIS = "ArcGIS",
+  BingMaps = "BingMaps",
+  MapboxImagery = "MapboxImagery",
+  TileURL = "TileURL",
+  WMS =  "WMS",
+  WMTS =  "WMTS"
+}
+
+/** Id of a map layer format.  In order to be displayed, a corresponding
+ * format object must have been registered in the [[MapLayerFormatRegistry]]
+ * @public
+ */
+export type MapLayerFormatId = string | ImageryMapLayerFormatId;
```

Comment:
@pmconne I think this the best balance between the flexibility of a loose string, and provide some guidance to API consumers.

That way it will still be possible to add a maplayer in a format that is still not "public".

Are you ok with that approach ?

Reply:
ok.

Reply:
Is the ImageryMapLayerFormatId enum new? If so please change it to a string union. String enums are redundant and verbose.

Reply:
I feel like string enums provide better 'intellisense' because you can use <enumName>.<value>, but ok I will take your word for it.

Reply:
I am told that competent editors supply auto-completion for string unions too. I won't argue about it though - do what you want with it.

Reply:
@pmconne
Maybe all we need is something similar to 'changeBackgroundMapProvider'.  Would it be a good idea to create 'ImageMapLayerSettings' from Props that takes a union type instead of formatId:string (But still keeping the formatId:string version).  Any suggestion on how that could be done without creating too much confusion ?